internal/testing/factories/factories.go

```diff
@@ -0,0 +1,50 @@
+package factories
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"golang.org/x/crypto/ssh"
+)
+
+var RSAGenerationError = errors.New("rsa key generation failed")
+
+var nameSequence = 1
+
+// GeneratePubkey generates a pubkey and stores it in the context stub
+func GeneratePubkey(ctx context.Context, pubkey models.Pubkey) (models.Pubkey, error) {
+	var err error
+	if pubkey.AccountID == 0 {
+		pubkey.AccountID = 1
+	}
```

Comment:
This is a side effect I would not expect and it will bite someone's ass pretty soon.